relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -375,6 +380,76 @@ fn extract_transaction_metrics_inner(
     Ok(())
 }
 
+fn extract_span_metrics(
+    aggregator_config: &AggregatorConfig,
+    event: &mut Event,
+    metrics: &mut Vec<Metric>, // output parameter
+) -> Result<(), ExtractMetricsError> {
+    // TODO(iker): measure the performance of this whole method
+
+    if event.ty.value() != Some(&EventType::Transaction) {
+        return Ok(());
+    }
+    let (Some(_), Some(&end)) = (event.start_timestamp.value(), event.timestamp.value()) else {
+        relay_log::debug!("failed to extract the start and the end timestamps from the event");
+        return Err(ExtractMetricsError::MissingTimestamp);
+    };
+
+    let Some(timestamp) = UnixTimestamp::from_datetime(end.into_inner()) else {
+        relay_log::debug!("event timestamp is not a valid unix timestamp");
+        return Err(ExtractMetricsError::InvalidTimestamp);
+    };
+
+    // Validate the transaction event against the metrics timestamp limits. If the metric is too
+    // old or too new, we cannot extract the metric and also need to drop the transaction event
+    // for consistency between metrics and events.
+    if !aggregator_config.timestamp_range().contains(&timestamp) {
+        relay_log::debug!("event timestamp is out of the valid range for metrics");
+        return Err(ExtractMetricsError::InvalidTimestamp);
+    }
+
+    if let Some(spans) = event.spans.value_mut() {
+        // Collect the shared tags for all the metrics and spans on this transaction
+        let mut shared_tags = BTreeMap::new();
+
+        if let Some(transaction_name) = event.transaction.value() {
+            shared_tags.insert("transaction".to_owned(), transaction_name.to_owned());
+        }
+        // TODO(iker): must add shared tags: environment, op, release
+
+        for annotated_span in spans {
+            if let Some(span) = annotated_span.value_mut() {
+                // TODO(iker): add span-specific tags here
```

Comment:
For tags we could use something similar to `extract_universal_tags`, to prepare the common tags for each incoming event.

Reply:
Yes, that's an improvement I decided to leave out of scope for now, and that we can iterate on later.